src/lib/utils/index.ts

```diff
@@ -19,22 +20,45 @@ const uploadApiEndpoint = `https://zns-backend.netlify.app/.netlify/functions/up
  * @param params Metadata parameters
  * @returns URI to the created Metadata
  */
-
-export const createDomainMetadata = async (params: DomainMetadataParams) => {
-	// upload image to http backend
-
+const getMetadataObject = async (params: DomainMetadataParams) => {
+	// upload images to http backend
 	const imageResponse = await fetch(uploadApiEndpoint, {
 		method: 'POST',
 		body: JSON.stringify(params.image),
 	});
 	const image = (await imageResponse.json()) as UploadResponseDTO;
 
+	if (params.previewImage!) {
```

Comment:
`params.previewImage` may be null here. Please do not use `!` like this (non-null assertion operator) unless absolutely necessary, it's a generally dangerous operator to use.

Instead a better way to do this check is:

```
if (params.previewImage) {
...
}
```